test/gtest/uct/ib/test_cqe_zipping.cc

```diff
@@ -0,0 +1,161 @@
+/**
+* Copyright (C) NVIDIA Corporation. 2022.  ALL RIGHTS RESERVED.
+*/
+
+#include <common/test.h>
+#include <common/test_helpers.h>
+#include <uct/ib/test_ib.h>
+
+extern "C" {
+#include <uct/api/uct.h>
+#include <uct/ib/base/ib_iface.h>
+}
+
+#define UCT_INSTANTIATE_MLX5_TEST_CASE(_test_case) \
+    _UCT_INSTANTIATE_TEST_CASE(_test_case, rc_mlx5) \
+    _UCT_INSTANTIATE_TEST_CASE(_test_case, dc_mlx5) \
+    _UCT_INSTANTIATE_TEST_CASE(_test_case, ud_mlx5)
+
+class test_cqe_zipping : public test_uct_ib_with_specific_port {
+public:
+    static bool check_counters()
+    {
+#if defined(ENABLE_STATS)
+        return !RUNNING_ON_VALGRIND;
+#else
+        return false;
+#endif
+    }
+
+    void send_while_possible()
+    {
+        while (am_zcopy() != UCS_ERR_NO_RESOURCE) {
+            ++m_send_cnt;
+        }
+    }
+
+    void wait_for_completion() const
+    {
+        /*
+         * Local deadline prevents hanging inside receive loop in case of
+         * connectivity issues
+         */
+        const ucs_time_t deadline = ucs::get_deadline(60);
+        while ((m_send_cnt != m_recv_cnt) && (ucs_get_time() < deadline)) {
+            progress();
+        }
+        ASSERT_EQ(m_send_cnt, m_recv_cnt);
+    }
+
+    size_t get_total_zipped_count() const
+    {
+#if defined(ENABLE_STATS)
+        constexpr int counters[2] = {UCT_IB_IFACE_STAT_TX_COMPLETION_ZIPPED,
+                                     UCT_IB_IFACE_STAT_RX_COMPLETION_ZIPPED};
+        size_t zipped_cqes_count  = 0;
+
+        for (auto &entity : m_entities) {
+            auto stats = ucs_derived_of(entity->iface(), uct_ib_iface_t)->stats;
+            for (auto &counter : counters) {
+                zipped_cqes_count += UCS_STATS_GET_COUNTER(stats, counter);
+            }
+        }
+        return zipped_cqes_count;
```

Comment:
minor: we could move `size_t zipped_cqes_count = 0;` and `return zipped_cqes_count;` outside the ifdef